### PR TITLE
AMP Cache SW: No IndexedDB

### PR DESCRIPTION
**Pull request by jridgewell**
The important stuff's happening in the [first commit](https://github.com/ampproject/amphtml/commit/91eb04a498c4e37db9ee03b17ef3420b7eed845e).

Eliminates our IndexedDB usage from the AMP Cache Service Worker. That also means we can remove our IndexedDB as Promised wrapper lib, making the service worker ~671 bytes (it was 2.94 kB before).

Fixes https://github.com/ampproject/amphtml/issues/4650.

### CODE REVIEW

File: src/service-worker/core.js [hunk 1]
```diff
@@ -107,36 +97,15 @@ const cachePromise = caches.open('cdn-js').then(result => {
...
-}));
-
-/**
  * Fetches the request, and stores it in the cache. Since we only store one
  * version of each file, we'll prune all older versions after we cache this.
  *
  * @param {!Request} request
+ * @param {string} requestFile the basename of the request
+ * @param {string} requestVersion the version of the request
+ * @return {!Promise<Response>}
```
Comment (aghassemi) [inline src/service-worker/core.js]: nit: I believe `fetch()` is guaranteed to have a non-null responsive in the resolve case.

File: src/service-worker/core.js [hunk 2]
```diff
@@ -150,41 +119,19 @@ function fetchAndCache(request) {
...
-            return version;
-          });
-        });
-      }).then(removedVersions => {
-        removedVersions.forEach(version => {
-          // Is the return value for the file-version we added (we explicitly
-          // returned an empty string for it)?
-          if (!version) {
-            return;
+      cache.keys().then(requests => {
```
Comment (aghassemi) [inline src/service-worker/core.js]: Update the block comment (line above stills says // Store the file version in IndexedDB0)

File: src/service-worker/core.js (hunk 1 above)
Comment (jridgewell) [inline src/service-worker/core.js]: Fixed.

File: src/service-worker/core.js (hunk 2 above)
Comment (jridgewell) [inline src/service-worker/core.js]: Fixed.

**Comment by aghassemi** [comment]
LGTM

**Comment by erwinmombay** [comment]
LGTM!

**Pull request merged**